### PR TITLE
CLOUDSTACK-9922 Unable to use 8081 port for Load balancing

### PR DESCRIPTION
**Pull request by SowjanyaPatha**
The default value for network.loadbalancer.haproxy.stats.port (global parameter) is configured as 8081.
After changing this parameter to other value eg:1234, even after the change not able to create Load balancer rules for the public port 8081,
Port 8081 is hard coded.

### CODE REVIEW

File: server/src/com/cloud/network/element/VirtualRouterElement.java [hunk 1]
```diff
@@ -171,6 +171,7 @@
     @Inject
     protected RouterDeploymentDefinitionBuilder routerDeploymentDefinitionBuilder;
 
+    static final ConfigKey<Integer> port= new ConfigKey<Integer>("Network",Integer.class, "network.loadbalancer.haproxy.stats.port", "8081", "Load Balancer(haproxy) stats port number.", true);
```
Comment (fmaximus) [inline server/src/com/cloud/network/element/VirtualRouterElement.java]: This key is already present in the Config enum class, which is legacy.
You might consider migrating the usages of the legacy enum value, to your new one.
Also that might mean, that you have to move it in VirtualRouterElementService interface
to make sure the plugins can use it.

**Comment by SowjanyaPatha** [comment]
@fmaximus Using the config enum class to access the global configuration and changed the method to non-static because it is used only once, so need not to be static.

**Comment by SowjanyaPatha** [comment]
@fmaximus can you please review it once.

**Comment by jayapalu** [comment]
LGTM

**Comment by vedulasantosh** [comment]
Test LGTM.
After changing the value of network.loadbalancer.haproxy.stats.port to XXXX. Able to create the load balancer rule with public port 8081.

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 810
 Hypervisor xenserver
 NetworkType Advanced
 Passed=107
 Failed=6
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_vm_snapshots.py

 * test_change_service_offering_for_vm_with_snapshots Failed

* test_volumes.py

 * test_06_download_detached_volume Failed

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 8 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 8 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 8 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 8 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_non_contigiousvlan.py
test_login.py
test_deploy_vm_iso.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by SowjanyaPatha** [comment]
tag:MergeReady

**Comment by cloudmonger** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 1018
 Hypervisor xenserver
 NetworkType Advanced
 Passed=101
 Failed=10
 Skipped=12

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/r2si930m8xxzavs/AAAzNrnoF1fC3auFrvsKo_8-a?dl=0

**Failed tests:**
* test_scale_vm.py

 * ContextSuite context=TestScaleVm>:setup Failing since 34 runs

* test_non_contigiousvlan.py

 * test_extendPhysicalNetworkVlan Failed

* test_deploy_vm_iso.py

 * test_deploy_vm_from_iso Failing since 73 runs

* test_list_ids_parameter.py

 * ContextSuite context=TestListIdsParams>:setup Failing since 47 runs

* test_volumes.py

 * test_06_download_detached_volume Failing since 2 runs

* test_vm_life_cycle.py

 * test_10_attachAndDetach_iso Failing since 73 runs

* test_routers_network_ops.py

 * test_01_isolate_network_FW_PF_default_routes_egress_true Failing since 106 runs

 * test_02_isolate_network_FW_PF_default_routes_egress_false Failing since 106 runs

 * test_01_RVR_Network_FW_PF_SSH_default_routes_egress_true Failing since 104 runs

 * test_02_RVR_Network_FW_PF_SSH_default_routes_egress_false Failing since 104 runs


**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_01_verify_libvirt
test_02_verify_libvirt_after_restart
test_03_verify_libvirt_attach_disk
test_04_verify_guest_lspci
test_05_change_vm_ostype_restart
test_06_verify_guest_lspci_again
test_static_role_account_acls
test_11_ss_nfs_version_on_ssvm
test_nested_virtualization_vmware
test_3d_gpu_support
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vm_snapshots.py
test_over_provisioning.py
test_global_settings.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_loadbalance.py
test_routers.py
test_reset_vm_on_reboot.py
test_deploy_vms_with_varied_deploymentplanners.py
test_network.py
test_router_dns.py
test_login.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_metrics_api.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_disk_offerings.py

**Comment by SowjanyaPatha** [comment]
tag:MergeReady

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 2]
```diff
@@ -2054,7 +2054,7 @@ public ServiceOffering createServiceOffering(final CreateServiceOfferingCmd cmd)
             if(!allowNetworkRate) {
                 throw new InvalidParameterValueException("Network rate can be specified only for non-System offering and system offerings having \"domainrouter\" systemvmtype");
             }
-            if(cmd.getNetworkRate().intValue() < 1) {
+            if(cmd.getNetworkRate().intValue() < 0) {
                 throw new InvalidParameterValueException("Failed to create service offering " + name + ": specify the network rate value more than 0");
```
Comment (wido) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: The error message below says something different then the if statement

Comment (SowjanyaPatha) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @wido incase of unlimited Network Rate we will give 0 value.

Comment (wido) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: @SowjanyaPatha Ok! I see

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by rhtyd** [comment]
@nvazquez @borisstoyanov can you review the code/results?

**Comment by rhtyd** [comment]
@jayapalu please avoid merging PRs and inform other people who may be still discussing the results and waiting for feedback, assessing them.

Also check before merging PRs that they have (1) appropriate JIRA ids in them, (2) the commits are squashed, ask the authors to do so or use github UI for squash/merging them, this PR was merged that created three commits and will make it challenging during release notes updationg: https://github.com/apache/cloudstack-docs-rn/tree/master/utils

**Pull request merged**